### PR TITLE
fix: KeyValue directive definition

### LINKED ISSUE #1359: fix `KeyValue` directive definition
`.tailcallrc.graphql` has the wrong definition for `KeyValues`. For eg:

Update the `gen_gql_schema.rs` so that it generates the correct .graphql and .json schema files.

**Current**
```graphql
directive @http(headers: KeyValues)
scalar KeyValues
```

**Expected**
```graphql
directive @http(headers: [KeyValue])

input KeyValue {
  key: String
  value: String
}
```

### PR DESCRIPTION
**Pull request by Bhavyajain21**
/claim #1359 
/fixes #1359 
**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **New Features**
	- Introduced a new input type for GraphQL schema to enhance data handling capabilities.

### CODE REVIEW

**Comment by Bhavyajain21** [comment]
please review @tusharmath

File: autogen/src/gen_gql_schema.rs [hunk 1]
```diff
@@ -660,6 +661,14 @@ fn write_all_input_types(
...
                 write_object_validation(writer, name, obj_valid, &defs, &mut new_extra_it)?
             }
+            ExtraTypes::KeyValue => {
+                writeln!(writer, "input KeyValue {{")?;
+                writer.indent();
+                writeln!(writer, "key: String")?;
+                writeln!(writer, "value: String")?;
+                writer.unindent();
+                writeln!(writer, "}}")?;
+            }
```
[the remark this answers is not in the document]
Comment (Bhavyajain21) [reply autogen/src/gen_gql_schema.rs:671]: OK, working on it

File: autogen/src/gen_gql_schema.rs [hunk 2]
```diff
@@ -684,6 +693,7 @@ fn generate_rc_file(file: File) -> Result<()> {
     let mut written_directives = HashSet::new();
 
     let mut extra_it = BTreeMap::new();
+    extra_it.insert("KeyValue".to_string(), ExtraTypes::KeyValue);
```
Comment (ssddOnTop) [inline autogen/src/gen_gql_schema.rs:696]: we can not insert it directly here

**Comment by ssddOnTop** [comment]
push the updated graphql file as well

**Comment by Bhavyajain21** [comment]
> push the updated graphql file as well

OK sure

File: autogen/src/gen_gql_schema.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bhavyajain21) [reply autogen/src/gen_gql_schema.rs]: Resolved

File: autogen/src/gen_gql_schema.rs (hunk 2 above)
Comment (Bhavyajain21) [reply autogen/src/gen_gql_schema.rs]: updated, please check

Comment (ssddOnTop) [reply autogen/src/gen_gql_schema.rs:699]: Please do not mark unresolved stuff as resolved

Comment (Bhavyajain21) [reply autogen/src/gen_gql_schema.rs:699]: Why we can't insert it directly here?

Comment (Bhavyajain21) [reply autogen/src/gen_gql_schema.rs:698]: updated, please check @ssddOnTop

**Comment by tusharmath** [comment]
closing in fav of https://github.com/tailcallhq/tailcall/pull/1372

**Pull request closed without merging**